driver/src/orderbook/streamed/orderbook.rs

```diff
@@ -64,6 +73,22 @@ impl Orderbook {
         });
     }
 
+    pub fn write_to_file(&self, path: impl AsRef<Path>) -> Result<()> {
+        // Write to tmp file until complete and then rename.
+        let mut temp_path = env::temp_dir();
```

Comment:
Instead of using `temp_dir` we should use the same path but with something like `.temp` appended. The problem with `temp_dir` is that we don't know where this directory is and renaming a file will fail if the two paths are not on the same filesystem.

Reply:
Great. Yes, I will do this. Although I tried this at first and got stuck not knowing how to concatenate strings... hopefully I can extend path or something.

Reply:
https://doc.rust-lang.org/std/path/struct.Path.html#method.with_extension could work.

Reply:
Managed to get the job done just like this:

```
let temp_path = path.as_ref().with_extension(".temp");
```